test/unit/codegen/codegen_llvm_ir.cpp

```diff
@@ -1283,6 +1286,37 @@ SCENARIO("Vector library calls", "[visitor][llvm][vector_lib]") {
     }
 }
 
+//=============================================================================
+// Fast math flags
+//=============================================================================
+
+SCENARIO("Fast math flags", "[visitor][llvm]") {
+    GIVEN("A function to produce fma and specified math flags") {
+        std::string nmodl_text = R"(
+            FUNCTION foo(a, b, c) {
+                foo = (a * b) + c
+            }
+        )";
+
+        THEN("instructions are generated with the flags set") {
+            std::string module_string =
+                run_llvm_visitor(nmodl_text,
+                                 /*opt=*/true,
+                                 /*use_single_precision=*/false,
+                                 /*vector_width=*/1,
+                                 /*vec_lib=*/"none",
+                                 /*fast_math_flags=*/{"nnan", "contract", "afn"});
+            std::smatch m;
+
+            // Check flags for produced 'fmul' and 'fadd' instructions.
+            std::regex fmul(R"(fmul nnan contract afn double %.*, %.*)");
+            std::regex fadd(R"(fadd nnan contract afn double %.*, %.*)");
```

Comment:
👍  - Ok, so without `reassoc`, this will be `fma` right?

Reply:
For fma, what we really need is a `contract` (`reassoc` is more general and hence can be more inaccurate when combining FP instructions differently) - that one is more or less "safe" and is used for LLVM backends to create fmas.

Reply:
I don't think there is fma in IR (i.e. it is not replaced by intrinsic). It is in SelectionDAG where fma nodes are created and subsequently converted to fma machine instructions (depending on the target)

Reply:
Example: https://godbolt.org/z/7Gae9fT69 (removing `-emit-llvm -S` will show assembly with `vfmadd132pd`, while IR  has `fmul`s and `fadd`s still.

Reply:
Oh yes, I meant fmul and fadd is later combined into fma !

All looks good.